dts: bindings: dma: Change the property names in the bindings

rruuaanng opened the pull request:
Change the property names in the bindings and DTS
to use hyphens(-) for separation instead of underscores(_).

jhedberg commented:
@rruuaanng this has failing CI checks that need fixing, likely because of changes to existing or new boards (xg23 & xg29) that were recently merged.

jhedberg commented:
@rruuaanng there are still CI build failures which seem valid

rruuaanng commented:
@jhedberg The problem seems to be in `dts/arm/silabs/efr32bg2x.dtsi` (All files, this is the only), it has `#dma-cells=<3>`, while the bindings require the constant to be `1`.
```
  "#dma-cells":
    const: 1
```
Should I change it to `1`? I'm not sure what effect this will have on the board using it.

rruuaanng commented:
Emm, it looks like I accidentally modified the cell properties during implementation.

fabiobaltieri commented on doc/releases/migration-guide-4.1.rst:
wrong file, 4.2

kartben reviewed (changes requested):
Needs to update the current migration guide

rruuaanng committed:
dts: bindings: dma: Change the property names in the DTS

Unify property names in bindings and DTS, using
hyphens (-) instead of underscores (_) as separators.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
doc: Add a new migration guide entry to DMA

Rename the nxp,a_on and dma_channels properties
to nxp,a-on and dma-channels in the devicetree
and bindings of the DMA subsystem.

Signed-off-by: James Roy <rruuaanng@outlook.com>
